aptos-move/aptos-vm/src/validator_txns/dkg.rs

```diff
@@ -5,132 +5,108 @@ use crate::{
-                Ok((
-                    VMStatus::MoveAbort(AbortLocation::Script, failure as u64),
-                    VMOutput::empty_with_status(TransactionStatus::Discard(StatusCode::ABORTED)),
-                ))
-            },
-            Err(Unexpected(vm_status)) => Err(vm_status),
+            Err(vm_status) => Err(vm_status),
         }
     }
 
-    fn process_dkg_result_inner(
+    fn process_dkg_transaction_inner(
         &self,
         resolver: &impl AptosMoveResolver,
         log_context: &AdapterLogSchema,
         session_id: SessionId,
-        dkg_node: DKGTranscript,
-    ) -> Result<(VMStatus, VMOutput), ExecutionFailure> {
-        let dkg_state = OnChainConfig::fetch_config(resolver)
-            .ok_or_else(|| Expected(MissingResourceDKGState))?;
-        let config_resource = ConfigurationResource::fetch_config(resolver)
-            .ok_or_else(|| Expected(MissingResourceConfiguration))?;
-        let DKGState { in_progress, .. } = dkg_state;
-        let in_progress_session_state =
-            in_progress.ok_or_else(|| Expected(MissingResourceInprogressDKGSession))?;
-
-        // Check epoch number.
-        if dkg_node.metadata.epoch != config_resource.epoch() {
-            return Err(Expected(EpochNotCurrent));
+        dkg_transaction: DKGTransactionData,
+    ) -> Result<(VMStatus, VMOutput), VMStatus> {
+        // Verify the dkg transaction before execution
+        if let Some(status) = self
+            .validate_dkg_validator_transaction(
+                ValidatorTransaction::DKG(dkg_transaction.clone()),
+                resolver,
+            )
+            .status()
+        {
+            return Err(VMStatus::Error {
+                status_code: status,
+                sub_status: None,
+                message: None,
+            });
         }
 
-        // Deserialize transcript and verify it.
-        let pub_params = DefaultDKG::new_public_params(&in_progress_session_state.metadata);
-        let transcript = bcs::from_bytes::<<DefaultDKG as DKGTrait>::Transcript>(
-            dkg_node.transcript_bytes.as_slice(),
-        )
-        .map_err(|_| Expected(TranscriptDeserializationFailed))?;
+        let function_name;
+        let args;
 
-        DefaultDKG::verify_transcript(&pub_params, &transcript)
-            .map_err(|_| Expected(TranscriptVerificationFailed))?;
+        match dkg_transaction.metadata.transaction_type {
+            DKGTransactionType::DKGMeta => {
+                function_name = SET_DKG_META;
+                args = vec![dkg_transaction.data_bytes.as_move_value()];
+            },
+            DKGTransactionType::PublicKeyShares => {
+                function_name = FINISH_WITH_DKG_RESULT;
+                args = vec![
+                    MoveValue::Signer(AccountAddress::ONE),
+                    dkg_transaction.data_bytes.as_move_value(),
+                ];
+            },
+        }
```

Comment:
I've suggested this in GH so the formatting will be off. Hopefully the idea is clear though.
```suggestion
        let (function_name, args) = match dkg_transaction.metadata.transaction_type {
            DKGTransactionType::DKGMeta => (SET_DKG_META, vec![dkg_transaction.data_bytes.as_move_value()]),
            DKGTransactionType::PublicKeyShares => (FINISH_WITH_DKG_RESULT,  vec![
                    MoveValue::Signer(AccountAddress::ONE),
                    dkg_transaction.data_bytes.as_move_value(),
                ]),
        }
```

Reply:
fixed